docs/how-to/hip_runtime_api/asynchronous.rst

```diff
@@ -0,0 +1,488 @@
+
+Streams enable the overlap of computation and data transfer, ensuring
+continuous GPU activity. 
+
+To create a stream, the following functions are used, each defining a handle
+to the newly created stream:
+
+- :cpp:func:`hipStreamCreate`: Creates a stream with default settings.
+- :cpp:func:`hipStreamCreateWithFlags`: Creates a stream, with specific
+  flags, listed below, enabling more control over stream behavior:
+
+  - ``hipStreamDefault``: creates a default stream suitable for most
+    operations. The default stream is a blocking operation.
+  - ``hipStreamNonBlocking``: creates a non-blocking stream, allowing
+    concurrent execution of operations. It ensures that tasks can run
+    simultaneously without waiting for each other to complete, thus improving
+    overall performance.
+
+- :cpp:func:`hipStreamCreateWithPriority`: Allows creating a stream with a
+  specified priority, enabling prioritization of certain tasks.
+
+The :cpp:func:`hipStreamSynchronize` function is used to block the calling host
+thread until all previously submitted tasks in a specified HIP stream have
+completed. It ensures that all operations in the given stream, such as kernel
+executions or memory transfers, are finished before the host thread proceeds.
+
+.. note::
+
+  If the :cpp:func:`hipStreamSynchronize` function input stream is 0 (or the
+  default stream), it waits for all operations in the default stream to
+  complete.
+
+Concurrent execution between host and device
+-------------------------------------------------------------------------------
+
+Concurrent execution between the host (CPU) and device (GPU) allows the CPU to
+perform other tasks while the GPU is executing kernels. Kernels can be launched
+asynchronously using ``hipLaunchKernelDefault`` with a stream, enabling the CPU
+to continue executing other code while the GPU processes the kernel. Similarly,
+memory operations like :cpp:func:`hipMemcpyAsync` can be performed
+asynchronously, allowing data transfers between the host and device without
+blocking the CPU.
+
+Concurrent kernel execution
+-------------------------------------------------------------------------------
+
+Concurrent execution of multiple kernels on the GPU allows different kernels to
+run simultaneously to maximize GPU resource usage. Managing dependencies
+between kernels is crucial for ensuring correct execution order. This can be
+achieved using :cpp:func:`hipStreamWaitEvent`, which allows a kernel to wait
+for a specific event before starting execution.
+
+Independent kernels can only run concurrently if there are enough registers
+and shared memory for the kernels. To enable concurrent kernel executions, the
+developer may have to reduce the block size of the kernels. The kernel runtimes
+can be misleading for concurrent kernel runs, that is why during optimization
+it is a good practice to check the trace files, to see if one kernel is blocking another
```

Comment:
Can we clarify what we mean by tracing here.
User might confuse it with ltrace etc. Also can we point user to the document which helps them trace so that they do not have to search for it.

Reply:
I am linking in the rocprof documnetation.